homeassistant/components/tplink/entity.py

```diff
@@ -217,6 +208,35 @@ def available(self) -> bool:
         return self.coordinator.last_update_success and self._attr_available
 
 
+class CoordinatedTPLinkFeatureEntity(CoordinatedTPLinkEntity, ABC):
+    """Common base class for all coordinated tplink feature entities."""
+
+    entity_description: TPLinkFeatureEntityDescription
+    _feature: Feature
+
+    def __init__(
+        self,
+        device: Device,
+        coordinator: TPLinkDataUpdateCoordinator,
+        *,
+        feature: Feature,
+        description: TPLinkFeatureEntityDescription,
+        parent: Device | None = None,
+    ) -> None:
+        """Initialize the entity."""
+        self.entity_description = description
+        super().__init__(device, coordinator, parent=parent, feature=feature)
+
+    def _get_unique_id(self) -> str:
+        """Return unique ID for the entity."""
+        key = self.entity_description.key
+        if key == PRIMARY_STATE_ID:
+            return legacy_device_id(self._device)
+
+        key = LEGACY_KEY_MAPPING.get(key, key)
```

Comment:
This probably deserves a comment on the logic.

Reply:
Done in https://github.com/home-assistant/core/pull/120064